astropy/nddata/nddata.py

```diff
@@ -192,12 +219,51 @@ def mask(self, value):
 
     @property
     def unit(self):
+        """
+        `~astropy.units.Unit`: Unit for the data, if any.
+        """
         return self._unit
 
     @property
     def wcs(self):
+        """
+        any type: wcs for the data, if any.
+
+        Even though nothing is enforced, using `~astropy.wcs.WCS` as ``wcs``
+        attribute is encouraged.
+        """
         return self._wcs
 
     @property
     def meta(self):
+        """
+        `dict`-like: Meta information, if any.
+        """
         return self._meta
+
+    @property
+    def uncertainty(self):
+        """
+        any type: Uncertainty in the data, if any.
+
+        Uncertainty *should* have an attribute ``uncertainty_type`` that is
```

Comment:
That's just a copy of my comments on your previous comment about this:

I share your opinion, but on the other hand that's mostly equivalent to enforcing the uncertainty being a `NDUncertainty`. Since having an attribute requires a customized class and having uncertainty propagation requires that this class has an interface that matches the workflow of `ArithmeticMixin`. Since I'm not nearly finished with the arithmetic and propagation (propagation should work, but there are some unresolved issues regarding unit comparison I need to investigate) this requires the end-user or package that uses the `NDData` to emulate `NDUncertainty` or use it. 

But in some cases one might just consider saving some uncertainty sources there (maybe a dict) or the subclass just assumes standard-deviation or variance and works with that. I think issueing a warning when there is no such attribute should be enough. That forces someone who works with `NDData` to either endure the `log.Info`s or use `NDUncertainty` (or something like it). Currently if one wants to use some simpler form like just a  numpy array or dict one needs to subclass NDData and overwrite the uncertainty setter. With the `should` it is inconvieniant to use something else but one does not need any programming skills to use it.

I can revert the behaviour back to `must` but I really tried not to enforce anything but to suggest it instead.

@mwcraig I just realized that there is already a PR which implements an ``UnknownUncertainty`` class (#3714) if you really do not like the *should* I could implement an adapted version of ``UnknownUncertainty``. I could alter the ``getter`` so that anyone using an unknown uncertainty is 'believing' ``NDData`` has saved their ``value`` but in truth it is something that subclasses from ``NDUncertainty`` with the corresponding properties and methods (but altered in a way that it cannot be propagated with anything not even with itself). That would actually be only be a minor change.